src/soorgeon/cli.py

```diff
@@ -112,41 +117,69 @@ def test(filename):
     $ soorgeon test path/to/notebook.ipynb
 
     """
-    if filename.lower().endswith(".ipynb"):
+    name, extension = splitext(filename)
+    directory = dirname(abspath(filename))
+    # save to {name}-soorgeon-test.ipynb by default
+    if not output_filename:
+        output_filename = join(directory, f"{name}-soorgeon-test.ipynb")
+    else:
+        output_filename = join(directory, output_filename)
+    if extension.lower() == '.py':
         nb = jupytext.read(filename)
         # convert ipynb to py and create a temp file in current directory
-        directory = dirname(abspath(filename))
-        with tempfile.NamedTemporaryFile(suffix=".py",
-                                                delete=True,
-                                                dir=directory) as temp_file:
+        with tempfile.NamedTemporaryFile(suffix=".ipynb",
+                                         delete=True,
+                                         dir=directory) as temp_file:
             jupytext.write(nb, temp_file.name)
-            _test(temp_file.name)
+            _test(temp_file.name, output_filename)
     else:
-        _test(filename)
+        _test(filename, output_filename)
 
 
-def _test(filename):
+def _test(filename, output_filename):
     try:
-        exec(open(filename).read())
-        click.echo(f"Finished executing {filename}, no error encountered")
-    except (ModuleNotFoundError, AttributeError, SyntaxError) as err:
+        pm.execute_notebook(filename, output_filename, kernel_name='python3')
```

Comment:
having more than one statement in the try clause can unexpected things if the second statement is what fails, it's better to do this:

```python
try:
    pm.execute...
except:
   ...
else:
   # show message that it executed successfully
```